src/backend/distributed/commands/call.c

```diff
@@ -217,3 +202,86 @@ CallFuncExprRemotely(CallStmt *callStmt, DistObjectCacheEntry *procedure,
 
 	return true;
 }
+
+
+/*
+ * ShardPlacementWhenColocatedWithReferenceTable decides on a placement for delegating
+ * a procedure call that reads from a reference table.
+ *
+ * If citus.task_assignment_policy is set to round-robin, we assign a different placement
+ * on consecutive runs. Otherwise the function returns the first placement available.
+ */
+static ShardPlacement *
+ShardPlacementWhenColocatedWithReferenceTable(CitusTableCacheEntry *cacheEntry)
```

Comment:
Maybe we can use a slightly different name here, because the word `when` makes it look like this is conditional, however we do the `isReferenceTable` check before calling this function and when we call this function it is already a reference table, so usage of a word like `for` instead of `when` could be better here and in the other method.

Reply:
How does the following sound:

2 seperate `ShardPlacementWhenColocatedWithReferenceTable()` implementations renamed to `ShardPlacementForProcedureColocatedWithReferenceTable` and `ShardPlacementForFunctionColocatedWithReferenceTable`

Similarly `ShardPlacementWhenColocatedWithDistTable()` renamed to `ShardPlacementForProcedureColocatedWithDistTable` and `ShardPlacementForFunctionColocatedWithDistTable`

These are quite different, and I'd like to keep them seperate.

Reply:
I guess we talked offline on this, ideally we can have only 2 methods, 1 for reference table and 1 for dist table.

Reply:
done.